src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -197,7 +197,7 @@ public void Enqueue(Event e)
                 break;
             } while (true);
 
-            Thread.Sleep(1);  // give EventPump thread a chance to process the event
+            Thread.Sleep(0);  // give EventPump thread a chance to process the event
```

Comment:
I'd also love if you could write a timing test which fails before the change and passes after the change. That will help keep us from getting into the very same kind of scenario again.